ethereum-rpc-retrofit/src/test/java/pm/gnosis/ethereum/rpc/retrofit/RetrofitEthereumRpcConnectorTest.kt

```diff
@@ -30,55 +29,55 @@ class RetrofitEthereumRpcConnectorTest {
         connector = RetrofitEthereumRpcConnector(api)
     }
 
-    @Test
+    @Test(expected = NotImplementedError::class)
     fun receipt() {
-        val expected = Observable.error<JsonRpcTransactionReceiptResult>(NotImplementedError())
+        val expected = NotImplementedError()
 
-        given(api.receipt(MockUtils.any())).willReturn(expected)
+        given(api.receipt(MockUtils.any())).willThrow(expected)
 
         assertEquals(expected, connector.receipt(request))
         then(api).should().receipt(request)
         then(api).shouldHaveNoMoreInteractions()
     }
 
-    @Test
+    @Test(expected = NotImplementedError::class)
     fun block() {
-        val expected = Observable.error<JsonRpcBlockResult>(NotImplementedError())
+        val expected = NotImplementedError()
 
-        given(api.block(MockUtils.any())).willReturn(expected)
+        given(api.block(MockUtils.any())).willThrow(expected)
 
         assertEquals(expected, connector.block(request))
         then(api).should().block(request)
         then(api).shouldHaveNoMoreInteractions()
     }
 
-    @Test
+    @Test(expected = NotImplementedError::class)
```

Comment:
If you do it this way, does it continue execution after it throws?

I would expect that not even the `assertEquals` is executed.

Reply:
It does not. So basically the assertion is implicit with the expected exception type. We basically miss the assertion as well of method calls, which I guess are also important. I could create an extension function that `try/catches/asserts`